deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -704,24 +706,30 @@ def wait(self) -> None:
                     partitions.append(part_to_copy)
             param.data = instrument_w_nvtx(torch.cat)(partitions).view(param.ds_shape)
             param.ds_status = ZeroParamStatus.AVAILABLE
-
-            for part_to_copy in partitions:
-                if not get_accelerator().is_synchronized_device():
-                    part_to_copy.record_stream(get_accelerator().current_stream())
+            if handle_dependency:
```

Comment:
Would the following not work?

```suggestion
            if handle_dependency and not get_accelerator().is_synchronized_device() :
```

Reply:
Thanks, it's better to move the condition outside the loop, modified.